### PR TITLE
Adding PAPI Placeholders about MissileWars

### PR DESCRIPTION
**Pull request by RedstoneFuture**
New placeholders for plugins with PlaceholderAPI support:
- `%missilewars_lobby_displayname_<lobby name or 'this'>%`
- `%missilewars_arena_displayname_<arena name or 'this'>%`
- `%missilewars_arena_missileamount_<arena name or 'this'>%`

### CODE REVIEW

File: missilewars-plugin/src/main/java/de/butzlabben/missilewars/MissileWars.java [hunk 1]
```diff
@@ -120,6 +121,11 @@ public void onEnable() {
             PreFetcher.preFetchPlayers(new StatsFetcher(new Date(0L), ""));
         }
 
+        // Small check to make sure that PlaceholderAPI is installed
+        if(Bukkit.getPluginManager().getPlugin("PlaceholderAPI") != null) {
```
Comment (daniel-naegele) [inline missilewars-plugin/src/main/java/de/butzlabben/missilewars/MissileWars.java]: Formatting (space between if and brackets). Please always format files before commiting (Strg + Alt + L on IntelliJ)

```suggestion
        if (Bukkit.getPluginManager().getPlugin("PlaceholderAPI") != null) {
```

File: missilewars-plugin/src/main/java/de/butzlabben/missilewars/MissileWars.java [hunk 2]
```diff
@@ -120,6 +121,11 @@ public void onEnable() {
...
+            new MissileWarsPlaceholder(this).register();
```
Comment (daniel-naegele) [inline missilewars-plugin/src/main/java/de/butzlabben/missilewars/MissileWars.java:126]: Maybe add Info Log Message that the placeholders were successfully registered.

File: missilewars-plugin/src/main/java/de/butzlabben/missilewars/util/MissileWarsPlaceholder.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    @Override
+    public boolean persist() {
+        return true;
+    }
+
+    @Override
+    public String onRequest(OfflinePlayer offlinePlayer, String params) {
+
+        if (params.endsWith("_this")) {
+            if (!offlinePlayer.isOnline()) return "ERROR: No a player!";
```
Comment (daniel-naegele) [inline missilewars-plugin/src/main/java/de/butzlabben/missilewars/util/MissileWarsPlaceholder.java]: Maybe change the message to something "player is not online"?

File: missilewars-plugin/src/main/java/de/butzlabben/missilewars/util/MissileWarsPlaceholder.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+            Player player = offlinePlayer.getPlayer();
+            Game playerGame = GameManager.getInstance().getGame(player.getLocation());
+            if (playerGame == null) return "ERROR: Wrong world!";
```
Comment (daniel-naegele) [inline missilewars-plugin/src/main/java/de/butzlabben/missilewars/util/MissileWarsPlaceholder.java]: Make message more precise, as lobbies can be in a specific area in a world (they could be in a world with a lobby, but in the lobby)

File: missilewars-plugin/src/main/java/de/butzlabben/missilewars/util/MissileWarsPlaceholder.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+                // %missilewars_arena_displayname_<arena name or 'this'>%
+                if (params.equalsIgnoreCase("arena_displayname_" + arena.getName())) {
+                    return arena.getDisplayName();
+                }
+
+                // %missilewars_arena_missileamount_<arena name or 'this'>%
+                if (params.equalsIgnoreCase("arena_missileamount_" + arena.getName())) {
+                    return Integer.toString(arena.getMissileConfiguration().getMissiles().size());
+                }
```
Comment (daniel-naegele) [inline missilewars-plugin/src/main/java/de/butzlabben/missilewars/util/MissileWarsPlaceholder.java]: The comments here are very good. Maybe you could add a new section in the README.md with a link to a new wiki page containing all possible placeholders.

**Review by daniel-naegele — CHANGES_REQUESTED** [review]

File: missilewars-plugin/src/main/java/de/butzlabben/missilewars/util/MissileWarsPlaceholder.java (hunk 5 above)
Comment (RedstoneFuture) [reply missilewars-plugin/src/main/java/de/butzlabben/missilewars/util/MissileWarsPlaceholder.java]: I have referred to the documentation in the README. We don't need to be more precise about the placeholders, I think.

File: missilewars-plugin/src/main/java/de/butzlabben/missilewars/util/MissileWarsPlaceholder.java [hunk 6]
```diff
@@ -77,7 +81,7 @@ public String onRequest(OfflinePlayer offlinePlayer, String params) {
             for (Arena arena : lobby.getArenas()) {
 
                 // %missilewars_lobby_displayname_<lobby name or 'this'>%
-                if (params.equalsIgnoreCase("lobby_displayname_" + lobby.getName())){
+                if (params.equalsIgnoreCase("lobby_displayname_" + lobby.getName())) {
```
Comment (daniel-naegele) [inline missilewars-plugin/src/main/java/de/butzlabben/missilewars/util/MissileWarsPlaceholder.java:84]: Good catch 👍

**Review by daniel-naegele — APPROVED** [review]

**Pull request merged**